substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1286,6 +1286,9 @@ pub mod pallet {
 			}
 
 			ensure!(!targets.is_empty(), Error::<T>::EmptyTargets);
+
+			targets.dedup();
```

Comment:
I would move this after the `targets.len() <= T::NominationsQuota::get_quota(ledger.active)` check so we are not deduplicating against an unnecessary long vec.